### DISCUSSION TITLE
Different results of same prediction

### DISCUSSION THREAD

**Discussion by com-network** [ourownstory/neural_prophet#301] [Q&A - get help using NeuralProphet] [answered]
Hello, I'm new to Python and forecasting as well. I installed neural_prophet 0.2.8 on Windows and I'm trying to run a basic prediction with the example data "retail sales". Here is my code:

```
from neuralprophet import NeuralProphet
import pandas as pd
import sys

df = pd.read_csv('retail_sales.csv')
m = NeuralProphet()
metrics = m.fit(df, freq="D")
future = m.make_future_dataframe(df, periods=3)
forecast = m.predict(future)
print(forecast)
```

However, each time I run the code, I get different results

```
          ds     y          yhat1 residual1          trend  season_yearly
0 2016-05-02  None  473125.744169       NaN  456836.637925   16289.090672
1 2016-05-03  None  475489.995617       NaN  456874.438612   18615.549219
2 2016-05-04  None  477766.152243       NaN  456912.203705   20853.957437

          ds     y          yhat1 residual1          trend  season_yearly
0 2016-05-02  None  472677.368223       NaN  456801.186433   15876.180677
1 2016-05-03  None  474741.100649       NaN  456838.702369   17902.401617
2 2016-05-04  None  476723.109556       NaN  456876.253899   19846.871229

          ds     y          yhat1 residual1          trend  season_yearly
0 2016-05-02  None  473781.169642       NaN  457165.987979   16615.182775
1 2016-05-03  None  476346.063719       NaN  457203.859854   19142.206090
2 2016-05-04  None  478819.054432       NaN  457241.731729   21577.309356
```

So what did I do wrong or am I being misunderstanding anything here? Any help would be much appreciated.

**Comment by ourownstory** [comment, marked as answer]
Hi @com-network 
First off, thank you for sharing a proper code snippet!
Second, you may find an anser here: #224 
tl;dr: this is expected behavior, but you can prevent it from happening by setting the seed (see linked documentation in #224).
